sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2376,7 +2376,18 @@ class Dataset[T] private[sql](
     if (remainingCols.size == allColumns.size) {
       toDF()
     } else {
-      this.select(remainingCols: _*)
+      val df = this.select(remainingCols: _*)
+      withAction("drop", df.queryExecution) { physicalPlan =>
```

Comment:
why do we need to copy? and also `drop` isn't an action. Also, looks we're going to lose existing plans.

Reply:
e.g.:

```scala
Seq((1, 2), (3, 4)).toDF("a", "b").write.parquet("/tmp/abc")
spark.read.parquet("/tmp/abc").drop("a").explain(true)
```

Reply:
Probably, @amanomer just used my example code in the jira. That's just an example code and I didn't carefully check it (I just copied/pasted it from the other part). I'm not currently sure that this is an issue to fix as I said in the jira: https://issues.apache.org/jira/browse/SPARK-30421